src/state/slices/opportunitiesSlice/resolvers/foxy/types.ts

```diff
@@ -0,0 +1,10 @@
+import type { UserStakingOpportunityBase } from '../../types'
+
+export type UserUndelegation = {
+  completionTime: number
+  undelegationAmountCryptoBaseUnit: string
+}
+
+export type FoxySpecificUserStakingOpportunity = UserStakingOpportunityBase & {
+  undelegations: UserUndelegation[]
```

Comment:
it may be worth adding a comment here to the effect that "undelegations" has a very specific meaning in cosmossdk parlance, but has been coopted here for FOXy as it's conceptually very similar

Reply:
fe6ceda956549d6b68336b80e500da76b9ef4d79